### PR TITLE
Implementation of the latest stable Netty4 version

### PR DESCRIPTION
**Pull request by valib**
@UniversalMediaServer/developers can we test the latest Netty 4 version if it is worth to be used? For me it works.

---

This change is [Reviewable](https://reviewable.io/reviews/universalmediaserver/universalmediaserver/902)

### CODE REVIEW

**Comment by SubJunk** [comment]
@valib cool, maybe we can release a beta version with this update to see how it works for people. Netty 4 always worked perfectly for me but others had issues, so I'm not able to test it reliably.

**Comment by Sami32** [comment]
@valib I was just thinking about that with the 5.2.1 version to see if freezing/timeouts will be here again...maybe not pertinent but hope doesn't kill anyone :-D
I will be happy to test it.

**Comment by SubJunk** [comment]
This is working well for me. Maybe we can release it as a 7.0.0 beta soon, what do you think?

**Comment by Sami32** [comment]
@valib @subjunk Look promising :-)

I meeted a warning in the LOG, but i will do more testing this week-end.

> WARN  2016-06-18 02:46:19.203 [main] **Failed to find a usable hardware address from the network interfaces**; using random bytes: 22:e9:f9:4f:86:54:0c:8d
> DEBUG 2016-06-18 02:46:19.203 [main] -Dio.netty.machineId: 22:e9:f9:4f:86:54:0c:8d (auto-detected)
[UMS-debug_Netty4_Failed.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/321529/UMS-debug_Netty4_Failed.txt)

**Comment by valib** [comment]
Rebased + merged to the latest master.

**Comment by valib** [comment]
@UniversalMediaServer/developers can you test it and evaluate if it is worth to be merged?

**Comment by Sami32** [comment]
@valib Any advices how to test it correctly ?

**Comment by Nadahar** [comment]
Hasn't there been endless problems each time Netty 4 has been merged? I don't think it's sufficiant to do a general test of this, the problems that has been experiences earler with Netty 4 must be tested specifically.

I don't know what those have been, can those of you that have been involved in this list what the problems have been?

I'm not so sure the problem is with Netty, it's probably our implementation - but that doesn't help if it we don't fix our implementation. If it is like I suspect, no new version of Netty will fix it.

**Comment by valib** [comment]
@Sami32 nothing special just test if renderer is properly recognized and browsing and playing media is without errors.
@Nadahar what do you mean with our implementation?

**Comment by Nadahar** [comment]
@valib Netty isn't a ready-made HTTP server - it's a network application framework. That means that the actual implementation is up to us, so any problems experienced when using it can be just as much because we're using it wrong or doing something else wrong. 

Just because a given code works in Netty 3.x, it doesn't mean that it will work in Netty 4.x. Maybe there's a bug in our code that is hidden/masked by the way 3.x works for example. The possibilities are endless, so without looking into exactly the problems that's used to fail, it's impossible to say if it works now.

**Comment by SubJunk** [comment]
This is the commit where we last rolled back Netty 4 https://github.com/UniversalMediaServer/UniversalMediaServer/commit/9652667b766e492071981b107d6352fa150a4422
The comments on it indicate that it caused 'corrupt' streaming on Sony Bluray

**Comment by Nadahar** [comment]
@SubJunk It was in 7.0.0 for a while as well, but was rolled back. Why was that?

**Comment by SubJunk** [comment]
@Nadahar there's an open issue for it on our tracker https://github.com/UniversalMediaServer/UniversalMediaServer/issues/598
Maybe we should follow up on that. Is there anything we are hoping to gain from updating to Netty 4? Is it better?

**Comment by Nadahar** [comment]
I have no idea if there's a particular reason to upgrade to Netty 4, but I guess the general idea of not being "left behind" by sticking to software that no longer get any attention carry some weight, although not too much in my view.

From what I can see in #598, @atamariya has already given us the likely solution to the problem. If that is implemented and this is tested with an affected renderer, that might be the best solution.

**Comment by SubJunk** [comment]
@valib thanks for this. Now we need to find someone to test the fix. I don't suppose @skeptical can spare a few minutes to test?

**Comment by Sami32** [comment]
```logtalk
TRACE 2017-08-28 04:24:13.281 [nioEventLoopGroup-3-1] 
io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1413)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1399)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes(AbstractByteBuf.java:1388)
	at io.netty.buffer.AbstractByteBuf.readBytes(AbstractByteBuf.java:870)
	at io.netty.buffer.CompositeByteBuf.readBytes(CompositeByteBuf.java:1798)
	at io.netty.buffer.CompositeByteBuf.readBytes(CompositeByteBuf.java:44)
	at io.netty.buffer.AbstractByteBuf.readBytes(AbstractByteBuf.java:878)
	at io.netty.buffer.CompositeByteBuf.readBytes(CompositeByteBuf.java:1793)
	at io.netty.buffer.CompositeByteBuf.readBytes(CompositeByteBuf.java:44)
	at net.pms.network.RequestHandlerV2.channelRead0(RequestHandlerV2.java:255)
	at net.pms.network.RequestHandlerV2.channelRead0(RequestHandlerV2.java:57)
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:102)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:438)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:284)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1342)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:934)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:134)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:645)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138)
	at java.lang.Thread.run(Unknown Source)
```
[debug.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1254970/debug.txt)

**Comment by Sami32** [comment]
@valib You have fixed it :+1: 

Other than testing it on Sony renderers, testing it with users using the V1 method, like @ExSport if i remember correctly, should be nice to have as well.
As if it fixed the V2 for them, the V1 could and should be removed, and at least i will see an advantage of such upgrade, as by now i still didn't feel any practical improvement

**Comment by valib** [comment]
@Sami32 the V1 method is untouched and not need testing. To remove the V1 method is up to @SubJunk to decide.

**Comment by Sami32** [comment]
@valib I was trying to say that user using HTTP V1 method should test this PR to see if this new V2 work for them

**Comment by SubJunk** [comment]
So I guess none of us have a device that was affected by the original bug... Maybe we should just merge this if it works for us? @UniversalMediaServer/developers opinions?

**Comment by Nadahar** [comment]
Has the bug been found and fixed in our code? If not, I think it's naive to think that a newer version of Netty should solve anything. That assumes that the problem is in Netty, which I find very unlikely.

I think the very minimum that should be done is to test it on a renderer that is known to have experiences issues. Even if we don't have one, it should be possible to find some guinea pig by asking for testers.

**Comment by SubJunk** [comment]
A code change has been made to fix the bug but has not been verified by manual testing

**Comment by Sami32** [comment]
@SubJunk I disagree. It need to be tested before.

**Comment by valib** [comment]
@UniversalMediaServer/developers I think that the best way to test it is to merge this PR to the 7.0.0 branch because there are some volunteers whose are testing this branch and maybe they could be able to give us some feedback. The other way is to merge this to the `master`as @SubJunk proposed and see what will happenned. @Sami32 I am not sure that there is anyone who can help us with this branch testing because it is here for the long time and nobody except developers contribute.

**Comment by Sami32** [comment]
@valib  I do not agree with you. No one did the effort or even considered make a build and ask to @ExSport port or @squadjot , that support UMS for long and are able to test it a build was offered to them, so i don't think that flaw come from users side as i think that they will be happy to help if asked, IMO.
Of course, it's only my point of view.

If you want merge it into 7.0.0, fine, no problem for me.

**Comment by valib** [comment]
@Sami32 I think that the @ExSport or @squadjot and others are following this PR but because they have  not the effected Sony TV they are not contributed.

**Comment by Sami32** [comment]
@valib And me i know that @squadjot have 2 Sony renderers and that @ExSport use the HTTP V1 method...

**Comment by Nadahar** [comment]
@valib I've tested PR's I've been working on by using the forum. You can't expect that users will just show up here and offer to help. You have to either post a question in some existing forum threads regarding the issue, or create a new thread where you ask for testers. You then need to make a build for them for the OS they use.

To me upgrading to Netty 4 is a bit difficult to understand. Sure, we'd want to be "on the latest version" as a general principle, but to me that isn't an argument that carry much weight. If there's something you need from a newer version, it's a different story.  I know of no problems that are caused by Netty 3 and no problems that would be solved by Netty 4, so the best case scenario is that everything works as today.

Netty 4 has been merged 2 (?) times already, and have had to be reversed. The problem is clearly on our side, not with Netty as it's used by a lot of projects and this would be a very fundamental bug that would have been solved. Despite having had no luck previously, there's a push to try to merge it again - without doing proper research and testing on the previously experienced issues.

To me this simply isn't logical, and I think it should be made absolutely sure that everything is working fine before merging it again. I simply don't see what makes it such a hurry. I think "testing" in master is a terrible idea, especially with things like this that can be hard to figure out.

**Comment by ExSport** [comment]
Hi all. Unfortunately my parents have PanTV only and also I never used HTTPv1 but v2 only.
If I remember well last time I tested on mother's new LG? But definitely not Sony one.
Also I little bit lost in thread so not sure what exactly should be check, what was initial problem :)

**Comment by Sami32** [comment]
@ExSport Thanks for answering. Sorry, i was sure that it was you that used the HTTP V1 method, I've misunderstood [your words](https://github.com/UniversalMediaServer/UniversalMediaServer/pull/462#issuecomment-71917546).
If we believe these posts, it seem that [@valib](https://github.com/UniversalMediaServer/UniversalMediaServer/pull/462#issuecomment-71828374) and [@SubJunk](https://github.com/UniversalMediaServer/UniversalMediaServer/issues/685#issuecomment-145433985) was already aware of the HTTP V1 method issues since precedent Netty implementation try.
It's why i suggested mandatoty tests with Sony renderers and with HTTP Engine V1 method users, to be sure to no merge again bugs and revert it later, and clean the V1 method if no issue is met anymore.

Damn...i cannot remember who use the V1 method.

**Comment by Nadahar** [comment]
@Sami32 I often use v1, although that's no "absolute" rule. I mostly only test stuff anyway. 

I'd like that add that there's nothing "wrong" or "old fashioned" with v1. v1 use the standard Java http server, but AFAICR it doesn't support chunked transfer - which is used by some renderers.

**Comment by Sami32** [comment]
@Nadahar I didn't knew about that V1 limitation, but AFAICU, chunked status seem also used to hide range and size values issues. We use chunked transfer, but some others media servers don't, for the same renderers. And when you look at our range and size values WOAH..

**Comment by squadjot** [comment]
If anyone needs testing on a old Sony TV W5500 , just let me know

**Comment by SubJunk** [comment]
@squadjot thanks for the offer. We really need someone whose device uses chunked transfer to test. That is most of the LG and Samsung TVs, and Sony Blu-Ray players

**Comment by SubJunk** [comment]
I just tested this with chunked transfer on my TV and it worked. I don't know if that is really good enough because my TV doesn't *need* chunked transfer. What do you guys think?

**Comment by Nadahar** [comment]
I've never gotten a clear understanding of the problem with Netty 4 (or more likely our use of it). Neither have I ever understood what the advantage is of upgrading, so until I have a better understanding of the situation I don't have an opinion about this. History suggests that there is a risk of problems using Netty 4 with our implementation, and I haven't yet seen any reward, so the only view I have is that it's not tempting to risk problems for nothing.

**Comment by SubJunk** [comment]
@Nadahar I'm not aware of any problems with Netty 3 that we experience that are fixed by Netty 4.1. Looking through their changelogs I see a lot of bugfixes but it is hard to know when the bugs themselves were introduced, maybe they have just been fixing bugs that were created in 4+.

@valib are you aware of any benefit we get from updating to Netty 4.1?

**Comment by valib** [comment]
@SubJunk benefits are described in http://netty.io/wiki/new-and-noteworthy-in-4.0.html which is improved in the version 4.1. Implementing `ByteBuf` in the version 4.0 improved the overall throughput around 5%. Is it worth enough?
In fact this PR doesn't use new features from the version 4.1 but I think that the UMS should use the latest versions of the dependencies which could improve the responsibility or solve some bugs.

**Comment by Nadahar** [comment]
> In fact this PR doesn't use new features from the version 4.1 but I think that the UMS should use the latest versions of the dependencies which could improve the responsibility or solve some bugs.

I think this is the true motivation. While I don't think using old outdated software is a goal, I don't share the belief that "newer is always better". There are plenty examples of things that has "evolved" the wrong way.

While the above is a good "default" rule, I don't think it's something that should get a lot of "weight". The blind belief that newer is better also conveniently skips the fact that all new bugs come from new code. If newer code was always better, bugs would die out by themselves. That's not the case in my experience.

I'm pretty sure that the performance gains are completely irrelevant for our use, since UMS is usually serving only a handful connections or less, often just one. When they talk about performance gains I'm pretty sure they're talking about for servers with thousands of connections. When it comes to transfer speeds I think UMS is limited to I/O speeds, that is, the lowest speed of the source (local or network), the speed of the network between UMS and the renderer and the speed of the transcoding if transcoding is involved. I don't think you will ever manage to create a situation where Netty's handling is the bottleneck.

I've "always" believed that there's nothing wrong with Netty 4. I suspect that our implementation isn't following the "rules" properly or does something wrong that Netty 3 (by coincidence/implementation) is more forgiving of than Netty 4. I thus believe that the implementation should be checked properly and the problems corrected before upgrading. I could be wrong though, but it's what I see as the most likely cause for the previous issues when upgrading.

I don't believe in the idea that the problem is on Netty's side and that this problem has gone undetected by everybody else for ages, so that if we just wait long enough it will have "fixed itself". I therefore think it's an unnecessary risk to upgrade and "hope for the best" when prior experience has been negative as long as something that's likely could have caused it is fixed in our code.

But if you guys wants to do the whole merge -> bug reports -> reverse routine in the off chance that the problems has magically disappeared, I won't stand in your way. I'll just say that I won't spend time troubleshooting any issues that arise from this.

**Comment by SubJunk** [comment]
I don't feel strongly about it either way haha so it's hard to say which way is the best!

**Comment by SubJunk** [comment]
@valib we have a user wanting to test if this fixes their problem, would you be able to resolve these conflicts?

**Comment by valib** [comment]
@SubJunk now it is ready for testing

**Comment by valib** [comment]
@SubJunk now it is ready for testing

**Comment by SubJunk** [comment]
@valib I'm interested in getting this merged. I have tested it and it seems good to me. I am thinking of maybe separating it from v10, because it might make it harder to identify bugs in the API code, so what do you think about it going out as 9.6.0?

**Comment by valib** [comment]
@SubJunk agree with you to release it in the 9.6.0

File: src/main/java/net/pms/network/HttpException.java [hunk 1]
```diff
@@ -1,9 +1,9 @@
 package net.pms.network;
 
-import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpResponseStatus;
 
 public class HttpException extends Exception {
-
+	private static final long serialVersionUID = -4952529888336004924L;
```
Comment (SubJunk) [inline src/main/java/net/pms/network/HttpException.java:6]: Is this line needed?

File: src/main/java/net/pms/network/RequestHandlerV2.java [hunk 2]
```diff
@@ -42,270 +54,145 @@
...
+			DLNA_GETCONTENTFEATURES,
+			DLNA_TIMESEEKRANGE,
+			DLNA_TRANSFERMODE,
+			HOST,
+			NT,
+			RANGE,
+			SID,
+			SOAPACTION,
+			TIMEOUT,
+			USER_AGENT,
```
Comment (SubJunk) [inline src/main/java/net/pms/network/RequestHandlerV2.java:98]:
```suggestion
		ACCEPT,
		ACCEPT_LANGUAGE,
		ACCEPT_ENCODING,
		CALLBACK,
		CONNECTION,
		CONTENT_LENGTH,
		CONTENT_TYPE,
		DATE,
		DLNA_GETCONTENTFEATURES,
		DLNA_TIMESEEKRANGE,
		DLNA_TRANSFERMODE,
		HOST,
		NT,
		RANGE,
		SID,
		SOAPACTION,
		TIMEOUT,
		USER_AGENT,
```

File: src/main/java/net/pms/network/HttpException.java (hunk 1 above)
Comment (valib) [reply src/main/java/net/pms/network/HttpException.java:6]: @SubJunk to use it is recommended by Eclipse -  `The serializable class HttpException does not declare a static final serialVersionUID field of type long`

**Comment by SubJunk** [comment]
@valib I can see some exceptions in the logs after this merge. I haven't noticed any actual bugs so it may not be important but here they are:

```
[main] direct buffer constructor: unavailable
java.lang.UnsupportedOperationException: Reflective setAccessible(true) disabled
	at io.netty.util.internal.ReflectionUtil.trySetAccessible(ReflectionUtil.java:31)
	at io.netty.util.internal.PlatformDependent0$4.run(PlatformDependent0.java:233)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:227)
	at io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:289)
	at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:92)
	at io.netty.channel.nio.NioEventLoop.newTaskQueue0(NioEventLoop.java:279)
	at io.netty.channel.nio.NioEventLoop.newTaskQueue(NioEventLoop.java:150)
	at io.netty.channel.nio.NioEventLoop.<init>(NioEventLoop.java:138)
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:146)
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:37)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58)
	at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:52)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:96)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:91)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:72)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:52)
	at net.pms.network.HTTPServer.start(HTTPServer.java:117)
	at net.pms.PMS.init(PMS.java:630)
	at net.pms.PMS.createInstance(PMS.java:917)
	at net.pms.PMS.main(PMS.java:1087)
```

and

```
[main] jdk.internal.misc.Unsafe.allocateUninitializedArray(int): unavailable
java.lang.IllegalAccessException: class io.netty.util.internal.PlatformDependent0$6 cannot access class jdk.internal.misc.Unsafe (in module java.base) because module java.base does not export jdk.internal.misc to unnamed module @409bf450
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Unknown Source)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at io.netty.util.internal.PlatformDependent0$6.run(PlatformDependent0.java:347)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at io.netty.util.internal.PlatformDependent0.<clinit>(PlatformDependent0.java:338)
	at io.netty.util.internal.PlatformDependent.isAndroid(PlatformDependent.java:289)
	at io.netty.util.internal.PlatformDependent.<clinit>(PlatformDependent.java:92)
	at io.netty.channel.nio.NioEventLoop.newTaskQueue0(NioEventLoop.java:279)
	at io.netty.channel.nio.NioEventLoop.newTaskQueue(NioEventLoop.java:150)
	at io.netty.channel.nio.NioEventLoop.<init>(NioEventLoop.java:138)
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:146)
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:37)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84)
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58)
	at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:52)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:96)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:91)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:72)
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:52)
	at net.pms.network.HTTPServer.start(HTTPServer.java:117)
	at net.pms.PMS.init(PMS.java:630)
	at net.pms.PMS.createInstance(PMS.java:917)
	at net.pms.PMS.main(PMS.java:1087)
```

**Comment by valib** [comment]
@SubJunk it seems like known problem in Netty started with Java 9. See https://github.com/netty/netty/issues/7769

> this is expected and just a debug statement...

**Comment by valib** [comment]
@SubJunk other solution for java 11 is here https://stackoverflow.com/questions/57885828/netty-cannot-access-class-jdk-internal-misc-unsafe which works also for Java 14

**Comment by SubJunk** [comment]
@valib ok I'm happy to leave it since it doesn't seem like something to worry about

**Comment by valib** [comment]
@SubJunk agree

**Pull request merged**